### PR TITLE
Quote input string in BeNullOrEmpty error

### PR DESCRIPTION
**Pull request by fflaten**
## PR Summary
Fix missing quotes when single string is passed to `Should -BeNullOrEmpty`.

/cc @johlju 

## PR Checklist

- [x] PR has meaningful title
- [x] Summary describes changes
- [x] PR is ready to be merged
  - If not, use the arrow next to `Create Pull Request` to mark it as a draft. PR can be marked `Ready for review` when it's ready.
- [x] Tests are added/update *(if required)*
- [ ] Documentation is updated/added *(if required)*

### CODE REVIEW

File: src/functions/assertions/BeNullOrEmpty.ps1 [hunk 1]
```diff
@@ -52,7 +53,8 @@ function Should-BeNullOrEmpty([object[]] $ActualValue, [switch] $Negate, [string
             $failureMessage = NotShouldBeNullOrEmptyFailureMessage -Because $Because
         }
         else {
-            $failureMessage = ShouldBeNullOrEmptyFailureMessage -ActualValue $ActualValue -Because $Because
+            $valueToFormat = if ($singleValue) { $expandedValue } else { $ActualValue }
```
Comment (johlju) [inline src/functions/assertions/BeNullOrEmpty.ps1:56]: This seems to affect non empty hashtables too. Shall we add a unit tests to verify the exception message when the hast table is not empty? So it too returns the expected message? There is an existing test at line 22-24 but it does not verify the exception message.

Comment (fflaten) [reply src/functions/assertions/BeNullOrEmpty.ps1:56]: We don't do any special formatting for hashtables. It only drops the outer array, which in a single-item case should be the array Pester wrapped pipeline-input in.

```ps1
@{ hello = "world" } | Should -BeNullOrEmpty
# Pester 5.3.3
Expected $null or empty, but got @(System.Collections.Hashtable).

# After PR
Expected $null or empty, but got System.Collections.Hashtable

Comment (johlju) [reply src/functions/assertions/BeNullOrEmpty.ps1:56]: Looks good.

**Review by nohwnd — APPROVED** [review]
